### PR TITLE
Adds campaigns controller 

### PR DESCRIPTION
**Pull request by itsjoekent**
#### What's this PR do?

Adds the campaigns controller standalone
#### How should this be reviewed?

Call the functions!
#### Any background context you want to provide?

See the previous PR https://github.com/DoSomething/phoenix/pull/7047#issuecomment-247443454
#### Relevant tickets

I don't think this fixes any issue actually, but relevant too / assists / unblocks 
#7041
#7040
#7042

### CODE REVIEW

File: lib/themes/dosomething/paraneue_dosomething/js/campaign/Controller.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+   *   - {boolean} isStaffPick - Defaults to false.
+   *   - {int} page - Defaults to 1.
+   *   - {boolean} removeDuplicateResults
+   *   - {boolean} sortByRelevance
+   *   - {boolean} removeSignups
+   * @return {Promise}
+   */
+  getCampaigns(args) {
+    const query = {
+      staff_pick: args['isStaffPick'] ? +args['isStaffPick'] : 0, // API requires 1/0 instead of true/false
```
Comment (weerd) [inline lib/themes/dosomething/paraneue_dosomething/js/campaign/Controller.js]: Use the approach @DFurnes [mentioned](https://github.com/DoSomething/phoenix/pull/7047#r79043289)`Number(args['isStaffPick'])`

and add a `@todo` referencing this [bug](https://github.com/DoSomething/phoenix/issues/7049) 😄

File: lib/themes/dosomething/paraneue_dosomething/js/campaign/Controller.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+      page: args['page'] || 1,
+    };
+
+    return this.apiClient.get('campaigns', query).then(response => {
```
Comment (weerd) [inline lib/themes/dosomething/paraneue_dosomething/js/campaign/Controller.js:33]: Maybe you don't need this `return` if also using a `return` inside the anonymous function dealing with the response?

File: lib/themes/dosomething/paraneue_dosomething/js/components/CampaignSlide.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  constructor(props) {
+    super(props);
+
+    this.controller = new CampaignController();
+  }
+
+  render() {
+
+  }
+}
```
Comment (weerd) [inline lib/themes/dosomething/paraneue_dosomething/js/components/CampaignSlide.js:19]: 💃 yay!

**Review by weerd — CHANGES_REQUESTED** [review]
Just a couple comments on some small things :)

File: lib/themes/dosomething/paraneue_dosomething/js/campaign/Controller.js (hunk 2 above)
Comment (itsjoekent) [reply lib/themes/dosomething/paraneue_dosomething/js/campaign/Controller.js:33]: Still getting used to promises, this worked!

**Comment by itsjoekent** [comment]
@weerd Addressed your review!

File: lib/themes/dosomething/paraneue_dosomething/js/campaign/Controller.js (hunk 2 above)
Comment (DFurnes) [reply lib/themes/dosomething/paraneue_dosomething/js/campaign/Controller.js:33]: Wait... sorry to roll in after the fact, but this definitely should have a `return` statement. I'm not even sure how it works without it (the outer `return` returns the Promise from `getCampaigns`, and the inner `return` returns the value to the next step in the Promise chain).

Comment (weerd) [reply lib/themes/dosomething/paraneue_dosomething/js/campaign/Controller.js:33]: No worries, so am I :)

**Comment by weerd** [comment]
¡OLÉ! :sheep: it!

**Review by weerd — APPROVED** [review]

File: lib/themes/dosomething/paraneue_dosomething/js/campaign/Controller.js (hunk 2 above)
Comment (itsjoekent) [reply lib/themes/dosomething/paraneue_dosomething/js/campaign/Controller.js:33]: 👀  well this is odd then

Comment (DFurnes) [reply lib/themes/dosomething/paraneue_dosomething/js/campaign/Controller.js:33]: How did you test this to verify that removing the `return` still worked? As far as I can tell, block functions don't have implicit returns (only bracketless one-liners do that).

Comment (itsjoekent) [reply lib/themes/dosomething/paraneue_dosomething/js/campaign/Controller.js:33]: @DFurnes apparently when i called the function it was still the old file, you're def right :facepalm:

Comment (itsjoekent) [reply lib/themes/dosomething/paraneue_dosomething/js/campaign/Controller.js:33]: i often get this, where i refresh quickly enough the build didnt finish, so i still have the old file

Comment (weerd) [reply lib/themes/dosomething/paraneue_dosomething/js/campaign/Controller.js:33]: count to 10? 😉 :trollface:

Comment (itsjoekent) [reply lib/themes/dosomething/paraneue_dosomething/js/campaign/Controller.js:33]: 👊

Comment (weerd) [reply lib/themes/dosomething/paraneue_dosomething/js/campaign/Controller.js:33]: I deserve that. haha

Comment (weerd) [reply lib/themes/dosomething/paraneue_dosomething/js/campaign/Controller.js:33]: Yeah, I forgot we're in the `getCampaigns()` method which should return the data. Alternatively you could put all the logic inside that anonymous function inside a separate method like `processCampaigns()` or something, and then just call that from within the anonymous function? Not a huge deal though.

**Comment by itsjoekent** [comment]
Looks good to me! 
screen shot 2016-09-16 at 10 12 12 am

**Comment by weerd** [comment]
Looking good 👍

**Review by DFurnes — APPROVED** [review]

**Pull request merged**